backend/functions/case-assignments/case.assignment.function.test.ts

```diff
@@ -1,243 +1,178 @@
+    const { camsHttpResponse, azureHttpResponse } = buildTestResponseSuccess(undefined, {
+      statusCode: HttpStatusCodes.CREATED,
+    });
+    jest
+      .spyOn(CaseAssignmentController.prototype, 'createTrialAttorneyAssignments')
+      .mockResolvedValue(camsHttpResponse);
 
-  test('returns bad request 400 when a role is not passed in the request', async () => {
-    const requestOverride = {
-      ...request,
-      body: {
-        caseId: '001-90-90123',
-        attorneyList: ['John Doe'],
-        role: '',
-      },
-    };
-
-    const expectedResponse = {
-      message:
-        'Invalid role for the attorney. Requires role to be a TrialAttorney for case assignment. Required parameter(s) role is/are absent.',
-      success: false,
-    };
-
-    const httpErrorSpy = jest.spyOn(httpResponseModule, 'httpError');
-    await httpTrigger(context, requestOverride);
-    expect(context.res.body).toEqual(expectedResponse);
-    expect(context.res.statusCode).toEqual(400);
-    expect(httpErrorSpy).toHaveBeenCalledWith(expect.any(AssignmentError));
-    expect(httpErrorSpy).not.toHaveBeenCalledWith(expect.any(UnknownError));
+    const request = createMockAzureFunctionRequest(defaultRequestProps);
+    const response = await handler(request, context);
+    expect(response).toEqual(azureHttpResponse);
   });
 
-  test('returns bad request 400 when a role of TrialAttorney is not passed in the request', async () => {
-    const requestOverride = {
-      ...request,
-      body: {
-        caseId: '001-90-90123',
-        attorneyList: ['John Doe'],
-        role: 'TrialDragon',
-      },
-    };
-
-    const expectedResponse = {
-      message:
-        'Invalid role for the attorney. Requires role to be a TrialAttorney for case assignment.',
-      success: false,
-    };
-
-    const httpErrorSpy = jest.spyOn(httpResponseModule, 'httpError');
-    await httpTrigger(context, requestOverride);
-    expect(context.res.body).toEqual(expectedResponse);
-    expect(context.res.statusCode).toEqual(400);
-    expect(httpErrorSpy).toHaveBeenCalledWith(expect.any(AssignmentError));
-    expect(httpErrorSpy).not.toHaveBeenCalledWith(expect.any(UnknownError));
-  });
+  const errorTestCases = [
+    ['caseId not present', '', 'TrialAttorney', 'Required parameter(s) caseId is/are absent.'],
+    ['invalid caseId format', '123', 'TrialAttorney', 'caseId must be formatted like 01-12345.'],
+    [
+      'role not present',
+      '001-90-90123',
+      '',
+      'Invalid role for the attorney. Requires role to be a TrialAttorney for case assignment. Required parameter(s) role is/are absent.',
+    ],
+    [
+      'invalid role',
+      '001-90-90123',
+      'TrialDragon',
+      'Invalid role for the attorney. Requires role to be a TrialAttorney for case assignment.',
+    ],
+  ];
+  test.each(errorTestCases)(
+    'should return proper error response for %s',
+    async (_caseName: string, caseId: string, role: string, message: string) => {
```

Comment:
Why are we even defining a caseName if we're not going to use it.  Is it just for reference sake so we understand what the test is for?   Oh... is that was is being used in the string template with %s?

Reply:
Yes, it is what is being printed so that the test name makes it clear which case is failing if it ever fails.